proto/p4/GenericTable.md

```diff
@@ -0,0 +1,108 @@
+# GenericTable example p4info
+
+Below is an example for realizing MulticastGroupEntry as a GenericTable.
+The key (MatchFields) comprise of only the group_id. 2 different ways of
+defining the data fields has been presented. Only 1 of them is required.
+
+```
+generic_tables {
+  generic_table_type_id : 145
+  generic_table_type_name : "MulticastGroup"
```

Comment:
Why is the "resource" type represented as both an int *and* a string? It might be useful to do a wildcard-read filtered on all generic_table entries of the same type. For that use case, have the type as an integer makes sense.

Reply:
I am not sure I understood the question here. Are you talking about why we need an ID and a name?

Reply:
Yes, my question was why do we need both an ID and a name? However, I just noticed that your GenericTable/GenericTableInstance uses the same design as Extern/ExternInstance in `p4info.proto`. So, it is starting to make more sense to me. 😄

Reply:
Yes. I will be including a new item called a table_category_type_name and ID as well to make it more confusing before the next meeting tomorrow. Working on the writeup